src/components/CreateTargetForm/index.js

```diff
@@ -0,0 +1,121 @@
+import useValidation from 'hooks/useValidation';
+import createTargetValidations from 'validations/createTargetValidations';
+import useCreateTargetForm from 'hooks/useCreateTargetForm';
+import useTextInputProps from 'hooks/useTextInputProps';
+import { BLACK } from 'constants/colors';
+import common from 'constants/commonStyles';
+import fonts from 'constants/fonts';
+import styles from './styles';
+
+const FIELDS = {
+  areaLength: 'areaLength',
+  title: 'title',
+};
+const { TARGET } = strings;
+
+const CreateTargetForm = ({ onSubmit, onShowTopics, onHide, topic }) => {
+  const validator = useValidation(createTargetValidations);
+
+  const { values, errors, handleValueChange, handleSubmit, handleBlur } = useForm(
+    {
+      onSubmit,
+      validator,
+      validateOnBlur: true,
+    },
+    [onSubmit],
+  );
+
+  const {
+    handleOnPress,
+    targetError,
+    targetStatus,
+    getTopicsError,
+    getTopicsStatus,
+    topicError,
+  } = useCreateTargetForm(topic, TARGET, handleSubmit);
+
+  const inputProps = useTextInputProps(handleValueChange, handleBlur, values);
+
+  const topicText = () => {
+    if (getTopicsError) return TARGET.topicsError;
+    if (getTopicsStatus === LOADING) return TARGET.loadingTopics;
+    return (topic && topic.label) || TARGET.topicPlaceholder;
+  };
+
+  return (
+    <>
+      <TouchableOpacity style={styles.mainContainer} onPress={onHide} />
+      <KeyboardAwareScrollView keyboardShouldPersistTaps="handled" style={styles.container}>
+        <View style={styles.inputContainer}>
+          <Input
+            label={TARGET.area}
+            testID="area-input"
+            keyboardType="numeric"
+            error={errors[FIELDS.areaLength]}
+            invalid={targetError || errors[FIELDS.areaLength]}
+            short={false}
+            placeholder={TARGET.areaPlaceholder}
+            placeholderTextColor={BLACK}
+            {...inputProps(FIELDS.areaLength)}
+          />
+        </View>
+        <View style={styles.inputContainer}>
+          <Input
+            label={TARGET.titleLabel}
+            testID="target-title-input"
+            error={errors[FIELDS.title]}
+            invalid={targetError || errors[FIELDS.title]}
+            short={false}
+            placeholder={TARGET.titlePlaceholder}
+            placeholderTextColor={BLACK}
+            {...inputProps(FIELDS.title)}
+          />
+        </View>
+        <View>
+          <Text style={[common.inputLabel, common.leftPadding]}>{TARGET.topicLabel}</Text>
+          <TouchableOpacity
+            style={[styles.topic, topicError && common.inputBorderError]}
+            onPress={onShowTopics}>
+            <Text style={[styles.topicInput, Platform.OS === 'android' && fonts.bold]}>
+              {topicText()}
+            </Text>
+          </TouchableOpacity>
+        </View>
+        <ErrorView errors={{ ...errors, targetError, topicError }} />
+        <Button
+          testID="save-target-button"
```

Comment:
Are you creating tests for your app? Or why are you setting the testID field?

Reply:
I'm not! Just left the one's from the base and updated its value just in case.